### PR TITLE
Azure monitor: Make sure alert rule editor is not enabled when template variables are being used

### LINKED ISSUE #40249: Testing Alert Rule gives error
**What happened**:
I've two grafana dashboards for one datasource is  "Azure SQL" and for other  it is "Azure log Analytics Workspace".
I've created the Alert Rules using grafana dashboard and while testing Alert Rule, I'm getting below error:

**request handler response error {request failed, status: 404 Not Found, body: {\"error\":{\"message\":\"The requested path does not exist\",\"code\":\"PathNotFoundError\",\"correlationId\":\"7b069314-2e3d-4ce2-8e71-499e1f20aff2\"}}}",**

Detailed error message is below:
```{
  "firing": true,
  "state": "pending",
  "conditionEvals": " = true",
  "timeMs": "-938.866ms",
  "error": "request handler response error {request failed, status: 404 Not Found, body: {\"error\":{\"message\":\"The requested path does not exist\",\"code\":\"PathNotFoundError\",\"correlationId\":\"7b069314-2e3d-4ce2-8e71-499e1f20aff2\"}}  A <nil> [] [] 0xc0015512f0}",
  "logs": [
    {
      "message": "Condition[0]: Query",
      "data": {
        "from": 1633850920950,
        "queries": [
          {
            "refId": "A",
            "model": {
              "appInsights": {
                "dimension": [
                  "none"
                ],
                "metricName": "select",
                "rawQuery": false,
                "rawQueryString": "",
                "timeColumn": "timestamp",
                "timeGrain": "auto",
                "valueColumn": ""
              },
              "azureLogAnalytics": {
                "query": "KubePodInventory\n| where TimeGenerated > ago(2m)\n| where ClusterName == \"$AKS\"\n| summarize arg_max(TimeGenerated, *) by ContainerName\n| project Name, ContainerName, Namespace\n| join kind = inner (\n    Perf\n    | where $__timeFilter(TimeGenerated)\n    | where ObjectName == 'K8SContainer'\n    | where CounterName == 'cpuUsageNanoCores'\n    | where InstanceName contains '$AKS'\n    | extend ContainerNameParts = split(InstanceName, '/')\n    | extend ContainerNamePartCount = array_length(ContainerNameParts)            \n    | extend PodUIDIndex = ContainerNamePartCount - 2, ContainerNameIndex = ContainerNamePartCount - 1 \n    | extend ContainerName = strcat(ContainerNameParts[PodUIDIndex], '/', ContainerNameParts[ContainerNameIndex])\n    | extend Time = format_datetime(TimeGenerated, \"yyyy-MM-dd HH:mm\")\n    | project Time, ContainerName, CounterValue\n) on ContainerName\n| summarize CoresPerNamespace=sum(CounterValue) by Time, Namespace\n| order by Time asc\n| project make_datetime(Time), Namespace, CoresPerNamespace / 1000000",
                "resource": "$Workspace",
                "resultFormat": "time_series"
              },
              "azureMonitor": {
                "aggOptions": [],
                "dimensionFilter": "*",
                "metricDefinition": "select",
                "metricName": "select",
                "metricNamespace": "select",
                "resourceGroup": "select",
                "resourceName": "select",
                "timeGrain": "auto",
                "timeGrains": [],
                "top": "10"
              },
              "queryType": "Azure Log Analytics",
              "refId": "A",
              "subscription": "XXXXXXXXXX"
            },
            "datasource": {
              "id": 1,
              "name": "Azure Monitor"
            },
            "maxDataPoints": 1500,
            "intervalMs": 200
          }
        ],
        "to": 1633851220950
      }
    }
  ]
}
```

**What you expected to happen**: 
I'm expecting that it should not throm error while testing Alert rule.

**How to reproduce it (as minimally and precisely as possible)**:
Create any grafana dashboard and test alert rule.

**Anything else we need to know?**:
i've tried running grafana locally, in azure vm and azure app service. And, all three setups gives the same error while test the alert rule.

**Environment**:
- Grafana version: 8.2
- Data source type & version: Azure SQl DB  and Azure log analytics workspace
- OS Grafana is installed on: windows

### PR DESCRIPTION
**Pull request by sunker**
**What this PR does / why we need it**:
Checks if the currently active subquery contains a variable. This is necessary because otherwise the alerting rule editor will be enabled. 

**Which issue(s) this PR fixes**:

Fixes #40249

### CODE REVIEW

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts [hunk 1]
```diff
@@ -131,6 +131,23 @@ export default class Datasource extends DataSourceApi<AzureMonitorQuery, AzureDa
...
+      return true;
+    }
+
+    let subQuery;
+    if (query.queryType === AzureQueryType.AzureMonitor) {
+      subQuery = JSON.stringify(query.azureMonitor);
+    } else if (query.queryType === AzureQueryType.LogAnalytics) {
+      subQuery = JSON.stringify(query.azureLogAnalytics);
+    } else if (query.queryType === AzureQueryType.AzureResourceGraph) {
+      subQuery = JSON.stringify([query.azureResourceGraph, query.subscriptions]);
```
Comment (andresmgot) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts:145]: can you add a test that covers the case of a variable in this `subscriptions`? and maybe move the condition to check `query.subscriptions` to it's own `if`

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts [hunk 2]
```diff
@@ -131,6 +131,23 @@ export default class Datasource extends DataSourceApi<AzureMonitorQuery, AzureDa
...
+    }
+
+    return !!(subQuery && this.templateSrv.variableExists(subQuery));
```
Comment (andresmgot) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts:148]: nit: you don't need to convert to a boolean with `!!`

**Review by andresmgot — APPROVED** [review]
LGTM just a couple of minor suggestions.

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts (hunk 1 above)
Comment (sunker) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts:145]: Yeah good catch!

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts (hunk 2 above)
Comment (sunker) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts:148]: I believe we do, but I guess it should be enough to convert just the subQuery. Will fix

**Pull request merged**